### PR TITLE
{compiler}[GCCcore/13.3.0] AOCC v4.2.0

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

Depends on easyblock PR https://github.com/easybuilders/easybuild-easyblocks/pull/3437

### CODE REVIEW

**Comment by Thyre** [comment]
There's an issue for the AOCC EasyBlock in regards to the Clang version. Here, we switch to just using 16 as the Clang version so that the sanity check passes. 

I would rather keep the exact version and instead update the EasyBlock. The changes are straightforward and can mostly be copied from the Clang EasyBlock. 

https://github.com/easybuilders/easybuild-easyblocks/issues/3429

**Comment by maxim-masterov** [comment]
@Thyre Makes sense, I've created a PR for the AOCC easyblock and updated the version of Clang in this easyconfig.

**Comment by Thyre** [comment]
Test report by @Thyre
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3458
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
Linux - Linux EndeavourOS UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor, 1 x NVIDIA NVIDIA GeForce RTX 3070, 560.35.03, Python 3.12.6
See https://gist.github.com/Thyre/b3bd3d8efaa0947d2093a57a1129dbc3 for a full test report.

**Comment by Thyre** [comment]
Test report by @Thyre
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3458
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
Linux - Linux EndeavourOS UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor, 1 x NVIDIA NVIDIA GeForce RTX 3070, 560.35.03, Python 3.12.6
See https://gist.github.com/Thyre/81f62f80d621517079e400ece690f1d5 for a full test report.

**Comment by Thyre** [comment]
Test report by @Thyre
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3458
**SUCCESS**
Build succeeded for 0 out of 0 (1 easyconfigs in total)
Linux - Linux EndeavourOS UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor, 1 x NVIDIA NVIDIA GeForce RTX 3070, 560.35.03, Python 3.12.6
See https://gist.github.com/Thyre/f04557b7935933f08565d94878e891a4 for a full test report.

**Comment by Thyre** [comment]
Test report by @Thyre
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3458
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
Linux - Linux EndeavourOS UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor, 1 x NVIDIA NVIDIA GeForce RTX 3070, 560.35.03, Python 3.12.6
See https://gist.github.com/Thyre/1250e9aa6274af284c0c82332852ce1f for a full test report.

File: easybuild/easyconfigs/a/AOCC/AOCC-4.2.0-GCCcore-13.3.0.eb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+name = 'AOCC'
+version = '4.2.0'
+
+homepage = 'https://developer.amd.com/amd-aocc/'
+description = "AMD Optimized C/C++ & Fortran compilers (AOCC) based on LLVM 13.0"
```
Comment (Thyre) [inline easybuild/easyconfigs/a/AOCC/AOCC-4.2.0-GCCcore-13.3.0.eb:5]: Description should be updated, since this version is now based on LLVM 16.0.3.

**Comment by Thyre** [comment]
Test report by @Thyre
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
Linux - Linux EndeavourOS UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor, 1 x NVIDIA NVIDIA GeForce RTX 3070, 560.35.03, Python 3.12.6
See https://gist.github.com/Thyre/957032cfb123d50a4200d5ea2ad7254b for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3

**Comment by Thyre** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 0 out of 1 (1 easyconfigs in total) jsczen3c1.int.jsc-zen3.fz-juelich.de - Linux Rocky Linux 9.4, x86_64, AMD EPYC-Milan Processor (zen3), Python 3.9.18 See https://gist.github.com/boegelbot/a5135d5dca1f85b8911d5a35faa0aa7d for a full test report.

Interesting, the checksum has changed? I'll look into this.

**Comment by Thyre** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 0 out of 1 (1 easyconfigs in total) jsczen3c1.int.jsc-zen3.fz-juelich.de - Linux Rocky Linux 9.4, x86_64, AMD EPYC-Milan Processor (zen3), Python 3.9.18 See https://gist.github.com/boegelbot/a5135d5dca1f85b8911d5a35faa0aa7d for a full test report.

That's weird. I tried downloading the sources manually and the SHA256 matched. 
```console
 jreuter@ZAM054  ~  wget https://download.amd.com/developer/eula/aocc/aocc-4-2/aocc-compiler-4.2.0.tar
--2024-09-26 12:39:52--  https://download.amd.com/developer/eula/aocc/aocc-4-2/aocc-compiler-4.2.0.tar
Resolving download.amd.com (download.amd.com)... 23.37.35.150
Connecting to download.amd.com (download.amd.com)|23.37.35.150|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 142189580 (136M) [application/x-tar]
Saving to: ‘aocc-compiler-4.2.0.tar’

aocc-compiler-4.2.0.tar                 100%[=============================================================================>] 135,60M  10,2MB/s    in 13s     

2024-09-26 12:40:06 (10,4 MB/s) - ‘aocc-compiler-4.2.0.tar’ saved [142189580/142189580]

 jreuter@ZAM054  ~  sha256sum aocc-compiler-4.2.0.tar
ed5a560ec745b24dc0685ccdcbde914843fb2f2dfbfce1ba592de4ffbce1ccab  aocc-compiler-4.2.0.tar
```

**Comment by branfosj** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="--force-download"

**Comment by Thyre** [comment]
The AOCC installation will be non-functional, due to the same reasons described in https://github.com/easybuilders/easybuild-easyconfigs/pull/21629.

**Comment by Thyre** [comment]
Test report by @Thyre
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/3480
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
Linux - Linux EndeavourOS UNKNOWN, x86_64, AMD Ryzen 7 7800X3D 8-Core Processor, 1 x NVIDIA NVIDIA GeForce RTX 3070, 560.35.03, Python 3.12.7
See https://gist.github.com/Thyre/e0c7a0e38b074cf9d4242a2ea92d5311 for a full test report.

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ jsc-zen3
EB_ARGS="--include-easyblocks-from-pr 3480"

**Comment by SebastianAchilles** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 3480"

**Review by SebastianAchilles — APPROVED** [review]
lgtm

**Comment by SebastianAchilles** [comment]
Going in, thanks @maxim-masterov!

**Pull request merged**